### ISSUE TITLE
[SMARTSCAN] [HOLD for payment 2023-09-27] [$500] HIGH: Receipt modal closes when API response updates image url

### ISSUE THREAD

**Issue by luacmartins** [Expensify/App#26828]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Request money via `Scan` option
2. Tap the report preview
3. Tap the request preview
4. Tap the receipt image
5. Verify that the modal opens
6. Wait for a bit and notice that the modal closes

Video:
https://github.com/Expensify/App/pull/26815#issuecomment-1707199834

## Expected Result:
Modal should remain open

## Actual Result:
Modal closes when API response updates the image URL

## Workaround:
Reopen the modal

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [x] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:**
**Reproducible in staging?:**
**Reproducible in production?:**
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation
**Expensify/Expensify Issue URL:**
**Issue reported by:**
**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~019f02e5a5c44adaa8</li>
        <li>Upwork Job ID: 1701050065977769984</li>
        <li>Last Price Increase: 2023-09-11</li>
<li>Automatic offers: </li>
<ul>
<li>tienifr | Contributor | 26663318</li>
</ul></li>
    </ul>
</details>

**Comment by luacmartins** [comment]
Thanks for the link @allroundexperts! It seems like we're treating this as a new feature. So I'll remove the bug label, add new feature, mark it external and bump it to weekly for now.

**Comment by luacmartins** [comment]
Still looking at proposals

**Comment by dylanexpensify** [comment]
@luacmartins do we think we'll have this completed by Fancy week?

**Comment by luacmartins** [comment]
@tienifr I think your proposal looks good. Maybe we could use the filename as the unique identifier?

**Comment by luacmartins** [comment]
Gonna assign you to the issue regardless since I don't think we need to hold on the above.

**Comment by bfitzexpensify** [comment]
Payment summary:

Contributor: @tienifr to be paid $1500 for work and #urgency bonus via Upwork
C+: @thesahindia to be paid $1500 for work and #urgency bonus via manual request

**Comment by bfitzexpensify** [comment]
Another friendly bump on the BZ checklist @thesahindia

**Comment by bfitzexpensify** [comment]
It looks like this has caused a regression - see https://github.com/Expensify/App/issues/28718#issuecomment-1744944301. Do you want to take a look at that @thesahindia?

I'll open the update for testing once we address the regression, since we likely want to make sure the steps cover both cases.

**Comment by bfitzexpensify** [comment]
Another friendly bump @thesahindia

**Comment by bfitzexpensify** [comment]
OK, looks like https://github.com/Expensify/App/issues/27824 might fix this - holding this issue until that is live.

**Comment by luacmartins** [comment]
Still waiting on https://github.com/Expensify/App/issues/27824

**Comment by bfitzexpensify** [comment]
Same update, still waiting on https://github.com/Expensify/App/issues/27824

**Comment by luacmartins** [comment]
Gonna move this to weekly while it's on hold

**Comment by bfitzexpensify** [comment]
Still on hold for the moment

**Comment by luacmartins** [comment]
still waiting on https://github.com/Expensify/App/issues/27824

**Comment by dylanexpensify** [comment]
[27824](https://github.com/Expensify/App/issues/27824) is merged and deployed! What's next here?

**Comment by dylanexpensify** [comment]
Actually, confirmed - this bug doesn't exist any longer. @bfitzexpensify feel free to complete the remaining steps!

**Comment by bfitzexpensify** [comment]
Sweet. We left this open to address a regression from the initial work done here. Payments here are all complete, the BZ checklist was also completed. Now that that's solved, we can close this one out.

**Comment by bfitzexpensify** [comment]
Sorry, missed your message the other day. Payment summary:

Contributor: @tienifr $500 paid via Upwork
C+: @thesahindia $500 to be paid via manual request

**Comment by JmillsExpensify** [comment]
$500 payment approved for @thesahindia based on comment above.
